File: GO/Cigueña.go
package main

import "fmt"

type Cigueña struct {
	BaseAves
}

// Constructor con parametros
func NewCigueña(nombre, color, pico string) *Cigueña {
	return &Cigueña{
		BaseAves: *NewBaseAves(nombre, color, pico),
	}
}

// Construtor sin parametros
func NewCigueñaSinParametros() *Cigueña {
	return &Cigueña{
		BaseAves: *NewBaseAves("Cigueña", "Blanco", "Conico"),
	}
}

// Constructor copia
func NewCigueñaCopia(copia *Cigueña) *Cigueña {
	return &Cigueña{
		BaseAves: *NewBaseAves(
			copia.Nombre,
			copia.Color,
			copia.Pico,
		),
	}
}

// Destructor
func (c *Cigueña) Close() error {
	fmt.Printf("Destructor...")
	return nil
}

// Metodo de la clase padre
func (c *Cigueña) PonerHuevo() string {
	return "Ponen entre 5 y 10"
}

// Metodos de las interfaces
func (c *Cigueña) Volar() string {
	return "La ave vuela."
}

func (c *Cigueña) Migrar() string {
	return "Migran por alimento, reproduccion y evitar el clima extremo."
}

func (c *Cigueña) Habitat() string {
	return "Bosques, Ecosistemas acuáticos..."
}

File: JAVASCRIPT/clases/Cigueña.js
import { Aves } from "../clases-abstractas/Aves.js"
import { IVolar } from "../interfaces/IVolar.js"
import { IMigrar } from "../interfaces/IMigrar.js"
import { IHabitat } from "../interfaces/IHabitat.js"

export class Cigueña extends Aves {
    // Constructor con parametros
    constructor(nombre = "Cigueña", color = "Blanco", pico = "Conico") {
        super(nombre, color, pico)
    }

    //Constructor copia 
    clone() {
        return new Cigueña(this.nombre, this.color, this.pico)
    }

    // Destructor
    destroy() {
        console.log("Destructor...")
        this.nombre = false
        this.color = false
        this.pico = false
    }

    // Metodo de la clase padre
    ponerHuevo() {
        console.log("Ponen entre 5 y 10")
    }

    // Metodo implementado de las interfaces
    volar() {
        console.log("La ave vuela.")
    }

    migrar() {
        console.log("Migran por alimento, reproduccion y evitar el clima extremo.")
    }

    habitat() {
        console.log("Bosques, Ecosistemas acuáticos...")
    }
}